src/layouts/JoinForm.astro

```diff
@@ -0,0 +1,145 @@
+---
+import Countries from "@data/Countries.mjs";
+---
+
+<script blocking="render" async is:inline>
+	const getElement = document.getElementById.bind(document);
+	const EMAIL_REGEX = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+	let token;
+	function setToken(newToken) {
+		token = newToken;
+	}
+
+	function submit() {
+		getElement("btn-submit").classList.add("is-loading");
+		getElement("error-token").classList.add("is-hidden");
+		getElement("error-submit").classList.add("is-hidden");
+		let ready = true;
+
+		if (getElement("name").value.length === 0) {
+			ready = false;
+			getElement("error-name").classList.remove("is-hidden");
+		}
+
+		if (!EMAIL_REGEX.test(getElement("email").value)) {
+			ready = false;
+			getElement("error-email").classList.remove("is-hidden");
+		}
+
+		if (getElement("country").value === "Please select") {
+			ready = false;
+			getElement("error-country").classList.remove("is-hidden");
+		}
+
+		if (!getElement("bylaws").checked) {
+			ready = false;
+			getElement("error-bylaws").classList.remove("is-hidden");
+		}
+
+		if (!getElement("privacy").checked) {
+			ready = false;
+			getElement("error-privacy").classList.remove("is-hidden");
+		}
+
+		if (!token) {
+			ready = false;
+			getElement("error-token").classList.remove("is-hidden");
+		}
+
+		if (ready) {
```

Comment:
It seems like you forgot to reset the button of any of the fields failed verification — currently it just gets stuck and spins forever.